src/Adapter/MSTest.CoreAdapter/Execution/TestMethodInfo.cs

```diff
@@ -171,7 +171,48 @@ public virtual TestResult Invoke(object[] arguments)
 
         internal void SetArguments(object[] arguments)
         {
-            this.arguments = arguments;
+            this.arguments = this.ResolveArguments(arguments);
+        }
+
+        internal object[] ResolveArguments(object[] arguments)
+        {
+            ParameterInfo[] parameterInfos = this.TestMethod.GetParameters();
+            int requiredParameterCount = 0;
+            foreach (var parameter in parameterInfos)
+            {
+                if (!parameter.IsOptional)
+                {
+                    requiredParameterCount++;
+                }
+            }
+
+            if (requiredParameterCount == parameterInfos.Length)
+            {
+                return arguments;
+            }
+
+            if (arguments == null)
+            {
+                return null;
+            }
+
+            if (arguments.Length < requiredParameterCount || arguments.Length > parameterInfos.Length)
```

Comment:
Please add explanatory comments wherever possible.

Reply:
added comments in https://github.com/Microsoft/testfx/pull/604/commits/65cd652b25348974bc42a8e3ae57416af708f23c